proxy/src/proxy.rs

```diff
@@ -1053,9 +1046,24 @@ pub fn neon_options(params: &StartupMessageParams) -> Option<String> {
     Some(options)
 }
 
-pub fn is_neon_param(bytes: &str) -> bool {
+pub fn neon_options_str(params: &StartupMessageParams) -> Option<String> {
+    neon_options(params).map(|options| {
+        #[allow(unstable_name_collisions)]
+        options
+            .iter()
+            .map(|(k, v)| format!("{}:{}", k, v))
+            .sorted() // we sort it to use as cache key
+            .intersperse(" ".to_owned())
+            .collect()
+    })
+}
+
+pub fn neon_option(bytes: &str) -> Option<(String, String)> {
     static RE: OnceCell<Regex> = OnceCell::new();
-    RE.get_or_init(|| Regex::new(r"^neon_\w+:").unwrap());
+    RE.get_or_init(|| Regex::new(r"^neon_(\w+):(.+)").unwrap());
 
-    RE.get().unwrap().is_match(bytes)
+    let cap = RE.get().unwrap().captures(bytes)?;
+    let k = cap.get(1)?.as_str().to_owned();
+    let v = cap.get(2)?.as_str().to_owned();
+    Some((k, v))
```

Comment:
1. there is no `extract` in our version of Regex. Is this ok to bump it?
2. Why do you think that it should panic here? We are checking that this particular option is a neon option, it's totally valid to fail and return None for all other options.

Reply:
1. I shall ask in the rust channel
2. The `re.captures(input)` function already returns None if there's no match. The regex crate guarantees that the captures are present if the regex matches (at least in this case since there's a static set of captures)

Reply:
2. ah, right, great.

Reply:
regex lib was bumped